docs/modules/integrate/pages/kafka-connect-connectors.adoc

```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
 
 == Fault Tolerance
 
-When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored in the Jet engine. For example, partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information. 
+When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored by the Jet engine.
+Example for such states are partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.
+
+== Parallelism and reconfigurations
+
+Kafka Connect Source uses `tasks.max` configuration property to determine parallelism.
+Even if pipeline's total parallelism is higher than `tasks.max` value, the `tasks.max` value takes precedence.
+It is prohibited to have pipeline's total parallelism (edge's local parallelism * number of nodes) lower than `tasks.max`.
+
+After creation of an instance of a SourceConnector, Hazelcast Jet asks it to provide `tasks.max` task configurations. If the
+SourceConnector returns less than `tasks.max` configs, the remaining processors will become no-op.
+
+When SourceConnector requests a reconfiguration, Hazelcast asks for a new set of configs and distributes them across the cluster.
```

Comment:
When a Source connector requests reconfiguration, Hazelcast requests the new configuration and distributes it across the cluster.

Reply:
"a new set of configs" is intentional; the source connector returns as many distinct configurations as user requests

Reply:
And the configuration sets make up the new configuration - in most circumstances, the plural of configuration is configuration. 

Can you clarify what you mean by distinct configurations? Multiple config files/changed lines within a single config file/both of these? That might help pin down the right phrase....

Reply:
Just for the record (I commented elsewhere): I meant a collection of configurations (set in terms of math), I just went full 3yo with my English and forgot about "configuration set" in other meaning.